app/controllers/get_involved_controller.rb

```diff
@@ -0,0 +1,94 @@
+
+    if Services.feature_toggler.use_recommended_related_links?(content_item["links"], request.headers)
+      content_item["links"]["ordered_related_items"] = content_item["links"].fetch("suggested_ordered_related_items", [])
+    end
+
+    @content_item = PresenterBuilder.new(
+      content_item,
+      content_item_path,
+      view_context,
+    ).presenter
+  end
+
+  def load_get_involved_data
+    @open_consultation_count = retrieve_given_document_type("open_consultation")['total']
+    @closed_consultation_count = retrieve_date_filtered_closed_consultations(12)
+    @next_closing_consultations = [retrieve_next_closing] 
+    @recently_opened_consultations = retrieve_new_consultations
+    @recent_consultation_outcomes = retrieve_consultation_outcomes
+    @take_part_pages = sort_take_part(retrieve_given_document_type("take_part")['results'])
+  end
+
+  def retrieve_given_document_type(document_type)
+    Services.publishing_api.get_content_items(document_type: document_type).to_hash
+  end
+
+  def retrieve_special(params)
+    Services.publishing_api.get_content_items(params).to_hash
+  end
+
+  # Aims to find a count of consultations closed in the last n months.
+  # Could feasibly be done via SearchAPI calls but there is no direct or indirect way to search by closing date
+  # there so we have to go around the houses a bit via a single large P-API call.
+  def retrieve_date_filtered_closed_consultations(months)
+    closed_count = 0
+
+    cl_docs = retrieve_special({document_type: 'closed_consultation', per_page: 500, fields: ['details']})['results']
+    cl_docs.sort_by! { |k| -k['details']['closing_date'] }.reverse!
+
+    cl_docs.each do |doc|
+      if DateTime.parse(doc['details']['closing_date']) > DateTime.now.prev_month(months)
+        closed_count += 1
+      else
+        break
+      end
+    end
+
+    return closed_count # hard return for clarity
+  end
+
+  def retrieve_next_closing
+    open_consults = retrieve_given_document_type("open_consultation")['results']
+    open_consults.sort_by! { |k| k['details']['closing_date'] }[0]
+  end
+
+  def retrieve_new_consultations
+    open_consults = retrieve_given_document_type("open_consultation")['results']
+    sorted_desc = open_consults.sort_by! { |k| k['details']['opening_date'] }.reverse!.values_at(0..2)
+    parse_organisation_acronyms(sorted_desc)
+  end
+
+  def retrieve_consultation_outcomes
+    closed_consults = retrieve_given_document_type("consultation_outcome")['results']
+    sorted_desc = closed_consults.sort_by! { |k| k['details']['closing_date'] }.reverse!.values_at(0..2)
+    parse_organisation_acronyms(sorted_desc)
+  end
+
+  def parse_organisation_acronyms(consultations)
+    consultations.each do |consultation|
+      org_acronyms = []
+      organisations = consultation['links']['organisations']
+      organisations.each do |org_id|
+        org_acronyms << Services.publishing_api.get_live_content(org_id).parsed_content['details']['acronym']
```

Comment:
A little concerned about the nested loops and an API call in the middle of that nest - how many requests does it end up making? But I can't really see an alternative either.

Reply:
Believe me if there was a better way... Might be able to do it with SearchAPI now that I'm in that life, but it'd still likely be a loop unless I have a bunch of OR tags in the search (which might be possible but I don't know until I look at it).

Reply:
This kind of thing should be done via [link expansion](https://github.com/alphagov/publishing-api/blob/master/docs/link-expansion.md). You can add `acronym` to the list of fields [that get included in the representation of the organisation in the `links`](https://github.com/alphagov/publishing-api/blob/63a973916fcca6062dbfe61c51cdb60b4c1526c0/lib/expansion_rules.rb#L82) of the consultation. Then no API calls needs to be made, and the content is already there.

Although now that the data itself is coming from the Search API this might be harder, perhaps a better solution would be to get Search API to expose that in the search results. I'm not super familiar with the Search API, but perhaps something like this: https://github.com/alphagov/search-api/blob/35ce835ddde96c5a7cbdab61f7beeda9daf3b03f/lib/govuk_index/presenters/expanded_links_presenter.rb#L80-L83

Reply:
@thomasleese now we're in the search API it's actually more viable as said linking is already done! - See https://www.gov.uk/api/search.json?filter_format=consultation (literally any result)

I posted my comment this morning before I gave a solid look at it, but my internet is frankly so crap that it took this long to post.